cli/src/command.rs

```diff
@@ -1,109 +1,156 @@
-use node_transaction_factory::RuntimeAdapter;
-use sc_cli::VersionInfo;
-use sc_service::Roles as ServiceRoles;
+	fn description() -> String {
+		env!("CARGO_PKG_DESCRIPTION").into()
+	}
 
-/// Parse command line arguments into service configuration.
-pub fn run<I, T>(args: I, version: VersionInfo) -> sc_cli::Result<()>
-where
-	I: Iterator<Item = T>,
-	T: Into<std::ffi::OsString> + Clone,
-{
-	let args: Vec<_> = args.collect();
-	let opt = sc_cli::from_iter::<Cli, _>(args.clone(), &version);
+	fn author() -> String {
+		env!("CARGO_PKG_AUTHORS").into()
+	}
 
-	let mut config = sc_service::Configuration::from_version(&version);
+	fn support_url() -> String {
+		"support@centrality.ai".into()
+	}
 
-	match opt.subcommand {
-		None => {
-			opt.run.init(&version)?;
-			opt.run.update_config(&mut config, load_spec, &version)?;
-			opt.run.run(config, service::new_light, service::new_full, &version)
-		}
-		Some(Subcommand::Inspect(cmd)) => {
-			cmd.init(&version)?;
-			cmd.update_config(&mut config, load_spec, &version)?;
-
-			let client = sc_service::new_full_client::<
-				cennznet_primitives::types::Block,
-				cennznet_runtime::RuntimeApi,
-				cennznet_executor::Executor,
-			>(&config)?;
-			let inspect = node_inspect::Inspector::<cennznet_primitives::types::Block>::new(client);
-
-			cmd.run(inspect)
-		}
-		// Some(Subcommand::Benchmark(cmd)) => {
-		// 	cmd.init(&version)?;
-		// 	cmd.update_config(&mut config, load_spec, &version)?;
-
-		// 	cmd.run::<cennznet_primitives::types::Block, cennznet_executor::Executor>(config)
-		// }
-		Some(Subcommand::Factory(cli_args)) => {
-			cli_args.shared_params.init(&version)?;
-			cli_args.shared_params.update_config(&mut config, load_spec, &version)?;
-			cli_args
-				.import_params
-				.update_config(&mut config, ServiceRoles::FULL, cli_args.shared_params.dev)?;
-
-			config.use_in_memory_keystore()?;
-
-			match ChainSpec::from(config.expect_chain_spec().id()) {
-				Some(ref c)
-					if c == &ChainSpec::Development
-						|| c == &ChainSpec::CennznetNikau
-						|| c == &ChainSpec::CennznetAzalea => {}
-				_ => return Err("Factory is only supported for development, nikau, and azalea testnet.".into()),
-			}
+	fn copyright_start_year() -> i32 {
+		2017
+	}
 
-			// Setup tracing.
-			if let Some(tracing_targets) = cli_args.import_params.tracing_targets.as_ref() {
-				let subscriber = sc_tracing::ProfilingSubscriber::new(
-					cli_args.import_params.tracing_receiver.into(),
-					tracing_targets,
-				);
-				if let Err(e) = tracing::subscriber::set_global_default(subscriber) {
-					return Err(format!("Unable to set global default subscriber {}", e).into());
-				}
-			}
+	fn load_spec(&self, id: &str) -> Result<Box<dyn sc_service::ChainSpec>, String> {
+		Ok(match id {
+			// "dev" => Box::new(chain_spec::dev::config()?),
+			// "azalea" => Box::new(chain_spec::azalea::config()?),
+			// "nikau" => Box::new(chain_spec::nikau::config()?),
```

Comment:
Can we enable these now?

Reply:
Yup I've got a different PR to enable these: https://github.com/cennznet/cennznet/pull/289